crate/operator/restore_backup.py

```diff
@@ -749,6 +780,32 @@ async def handle(  # type: ignore
         :param name: The CrateDB custom resource name defining the CrateDB cluster.
         :param logger: the logger on which we're logging
         """
+
+        # Cloning a cluster will result in the destruction of all target cluster users.
```

Comment:
It was there before, but @tomach suggested to move it here, which makes sense because the ``RestoreSystemUserPasswordSubHandler`` gets executed even for errors

Reply:
I thought the same at first, but the `RestoreSystemUserPasswordSubHandler` runs even if the restore failed, whereas the admin username update we only want to do if everything was successful, right?

Reply:
ok cool, understood.